### ISSUE TITLE
Is there a way to count approvals made during the lifetime of the PR commits. 

### ISSUE THREAD

**Issue by kandelvijaya** [zalando/zappr#434]

**Comment by prayerslayer** [comment]
Hi, I don't understand. Could you elaborate and give examples please?

**Comment by kandelvijaya** [comment]
I create a PR. My teammates review it. Initially i have 5 commits. I get 1 approval. Then down the line i had to create another commit on the same PR maybe due to merge conflict. My system is set to allow 2 approvals. After the conflict resolution, i need to ask for 2 PR again instead of 1.

**Comment by kandelvijaya** [comment]
Im wondering it would be nice to have a feature that would count all the approvals gained during the PR not only the ones after the last commit. Does that make sense to you?

**Comment by j0k3r** [comment]
Well the last commit might breaks everything in the PR and so previous reviewers will need to re-review what you've done.

**Comment by prayerslayer** [comment]
Ah, okay. No unfortunately that's not what we consider good practice. You could have made mistakes during conflict resolution (happened to me certainly more than once). You could have completely changed your implementation after first merge proposal (also did that a couple of times). You could introduce an evil bug in your last commit. There are many situations in which you wanted to have an additional pair of eyes...

Basically what you're proposing is that once you have all approvals, you can do what you want on the branch? :grin:

**Comment by kandelvijaya** [comment]
it makes sense but for us, we have a big team of 14 people cramming, pulling and merging as you know. I wait for approval. Get 1. Someone comes along and merges his/her PR into develop. My PR branch is outdated. Github tells me to update. I update. Then again i need to ask for all the PR approvals.

I propose to atleast not remove approvals count after updating the PR branch with the develop. For commits that are pushed by user yes your opinion makes sense.

**Comment by prayerslayer** [comment]
You don't need to have your PR branch up to date with `master`, this you can change in the branch protection settings. (But we do have a discussion going on that these merges shouldn't reset the approvals: #310)

Will close as this seems to be solved, please reopen if you have further questions.
